src/test/java/gov/healthit/chpl/aqa/stepDefinitions/SurveillanceComplaintsSteps.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Fill data in complaint form.
+     * @param acb is ACB for which complaint is being entered
+     * @param receivedDate is received date of complaint
+     * @param closedDate is date when complaint was closed
+     * @param acbComplaintId ACB complaint ID
+     * @param oncComplaintId ONC complaint ID
+     * @param complainantType is type of complainant
+     * @param summary is summary of a complaint
+     */
+    @When("^I enter complaint data in fields: \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\" \"([^\"]*)\"$")
+    public void enterDataInComplaintFields(final String acb, final String receivedDate, final String closedDate,
+            final String acbComplaintId, final String oncComplaintId, final String complainantType,
+            final String summary) {
+        SurveillanceComplaintsPage.oncACBField(getDriver()).sendKeys(acb);
+        SurveillanceComplaintsPage.receivedDate(getDriver()).sendKeys(receivedDate);
+        SurveillanceComplaintsPage.closedDate(getDriver()).sendKeys(closedDate);
+        SurveillanceComplaintsPage.oncACBComplaintId(getDriver()).sendKeys(acbComplaintId);
+        SurveillanceComplaintsPage.oncComplaintId(getDriver()).sendKeys(oncComplaintId);
+        SurveillanceComplaintsPage.complainantType(getDriver()).sendKeys(complainantType);
+        SurveillanceComplaintsPage.summary(getDriver()).sendKeys(summary);
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.complainantContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.developerContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.oncATLContactedCheckbox(getDriver()));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.flagForONCReviewCheckbox(getDriver()));
+    }
+
+    @When("^I click save button$")
+    public void clickSaveButton() {
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", SurveillanceComplaintsPage.saveButtonOnNewComplaintForm(getDriver()));
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.complaintsTable(getDriver())));
+    }
+
+    /**
+     * Edit a complaint.
+     * @param acbComplaintId is a ACB complaint ID of a complaint to be edited
+     */
+    @When("^I click Edit complaint for ONC-ACB Complaint ID \"([^\"]*)\"$")
+    public void clickEditComplaint(final String acbComplaintId) {
+        getDriver().navigate().refresh();
+        getWait().until(ExpectedConditions.visibilityOf(SurveillanceComplaintsPage.complaintsTable(getDriver())));
+
+        List<WebElement> acbComplaintIdValue = getDriver().findElements(By.xpath("//*[@id=\"main-content\"]/div/div/div/div/table/tbody/tr/td[4]"));
```

Comment:
Shouldn't XPaths be in the Page Object files, not in the Steps?

Reply:
That's findElements, to find list of web elements (all acb complaint IDs) in the ONC-ACB Complaint Id column. I can't use single/specific element locator here as I am checking table data in ONC-ACB Complaint Id column row by row and then looking to click delete button based on value found in ONC-ACB Complaint Id column